tests/bgp/test_bgp_multipath_relax.py

```diff
@@ -0,0 +1,120 @@
+import pytest
+import logging
+
+logger = logging.getLogger(__name__)
+
+def get_t2_neigh(testbed):
+    dut_t2_neigh = []
+    for vm in testbed['topo']['properties']['topology']['VMs'].keys():
+        if 'T2' in vm:
+            dut_t2_neigh.append(vm)
+    return  dut_t2_neigh
+
+def get_t0_neigh(testbed, topo_config):
+    """
+    get all t0 router names which has vips defined 
+    """
+    dut_t0_neigh = []
+    for vm in testbed['topo']['properties']['topology']['VMs'].keys():
+        if 'T0' in vm:
+            if topo_config[vm].has_key('vips'):
+                dut_t0_neigh.append(vm)
+    return dut_t0_neigh
+
+def get_vips_prefix(dut_t0_neigh, topo_config):
+    vips_prefixes = []
+
+    # find all vips prefixes
+    for neigh in dut_t0_neigh:
+        prefixes = topo_config[neigh]['vips']['ipv4']['prefixes']
+        for prefix in prefixes:
+            if prefix not in vips_prefixes:
+                vips_prefixes.append(prefix)
+
+    # use the first prefix for testing
+    return vips_prefixes[0]
+
+def get_vips_prefix_paths(dut_t0_neigh, vips_prefix, topo_config):
+    vips_t0 = []
+    vips_asn = []
+
+    for neigh in dut_t0_neigh:
+        if vips_prefix in topo_config[neigh]['vips']['ipv4']['prefixes']:
+            vips_t0.append(topo_config[neigh]['bgp']['asn'])
+            vips_asn.append(topo_config[neigh]['vips']['ipv4']['asn'])
+    return vips_t0, vips_asn
+
+def get_bgp_v4_neighbors_from_minigraph(duthost, testbed):
+    mg_facts = duthost.minigraph_facts(host=duthost.hostname)['ansible_facts']
+    
+    # Find all V4 bgp neighbors from minigraph
+    bgp_v4nei = {}
+    for item in mg_facts['minigraph_bgp']:
+        if ':' in item['addr']:
+            continue
+        bgp_v4nei[item['name']] = item['addr']
+    return bgp_v4nei
+
+def test_bgp_multipath_relax(testbed, duthost):
+
+    logger.info("Starting test_bgp_multipath_relax on topology {}".format(testbed['topo']['name']))
+    topo_config = testbed['topo']['properties']['configuration']
+
+    bgp_v4nei = get_bgp_v4_neighbors_from_minigraph(duthost, testbed)
+
+    logger.info("bgp_v4nei {}".format(bgp_v4nei))
+    
+    # get all t0 routers name which has vips defined
+    dut_t0_neigh = get_t0_neigh(testbed, topo_config)
+
+    # get t2 neighbors
+    dut_t2_neigh = get_t2_neigh(testbed)
+    
+    if not dut_t0_neigh:
+        pytest.fail("Didn't find multipath t0's")
+
+    vips_prefix = get_vips_prefix(dut_t0_neigh, topo_config)
+    
+    logger.info("vips_prefix = {}".format(vips_prefix))
+
+    # find all paths of the prefix for test 
+    vips_t0, vips_asn = get_vips_prefix_paths(dut_t0_neigh, vips_prefix, topo_config)
+    
+    logger.info("vips_t0: {}, vips_asn: {}".format(vips_t0, vips_asn))
+
+    if len(vips_t0) <= 1:
+        pytest.fail("Did not find preconfigured multipath for the vips prefix under test")
```

Comment:
can you please unify the use of assert and pytest.fail. I'd like to mention that Pytest has a method to assert called [pytest_assert](https://github.com/Azure/sonic-mgmt/blob/master/tests/common/helpers/assertions.py#L3)

Reply:
@tahmed-dev  Changed the assert to pytest_assert in the latest commit